src/lib.rs

```diff
@@ -145,6 +156,47 @@ impl<'data> Iterator for DataFormatDescriptorIterator<'data> {
     }
 }
 
+struct KeyValueDataIterator<'data> {
+    data: &'data [u8],
+}
+
+impl<'data> Iterator for KeyValueDataIterator<'data> {
+    type Item = (&'data str, &'data [u8]);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut offset = 0;
+
+        let length = bytes_to_u32(self.data, &mut offset).ok()?;
+
+        let key_and_value = &self.data[offset..offset + length as usize];
```

Comment:
This will panic if `length` is garbage.
```suggestion
        let key_and_value = self.data.get(offset..offset.checked_add(length as usize)?)?;
```